From e4eb2d034631575ec8002c47d0263fc8f2eed7e6 Mon Sep 17 00:00:00 2001
From: Vijaya Kumar T M <vtmuni@codeaurora.org>
Date: Wed, 7 Jan 2015 14:28:41 +0530
Subject: [PATCH 148/176] QCamera2: Add param lock in getParameters

Not using param lock in getParameters leads
to concurrent access of the parameters memory
and a crash subsequently.

Change-Id: Ib380d587db510461c1bd85a30cdedfd67704677
---
 QCamera2/HAL/QCamera2HWI.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/QCamera2/HAL/QCamera2HWI.cpp b/QCamera2/HAL/QCamera2HWI.cpp
index 606543b..cb8b87a 100644
--- a/QCamera2/HAL/QCamera2HWI.cpp
+++ b/QCamera2/HAL/QCamera2HWI.cpp
@@ -2434,6 +2434,7 @@ char* QCamera2HardwareInterface::getParameters()
 
     int cur_width, cur_height;
 
+    pthread_mutex_lock(&m_parm_lock);
     //Need take care Scale picture size
     if(mParameters.m_reprocScaleParam.isScaleEnabled() &&
         mParameters.m_reprocScaleParam.isUnderScaling()){
@@ -2466,6 +2467,7 @@ char* QCamera2HardwareInterface::getParameters()
         pic_size.append(buffer);
         mParameters.set(CameraParameters::KEY_PICTURE_SIZE, pic_size);
     }
+    pthread_mutex_unlock(&m_parm_lock);
     return strParams;
 }
 
-- 
1.8.3.1

